src/Core/AuthenticationHelpers/ClientRoleHeaderAuthenticationMiddleware.cs

```diff
@@ -57,23 +59,41 @@ public ClientRoleHeaderAuthenticationMiddleware(RequestDelegate next,
     /// <param name="httpContext">Request metadata</param>
     public async Task InvokeAsync(HttpContext httpContext)
     {
+        // Determine the authentication scheme to use based on dab-config.json.
+        // Compatible with both ConfigureAuthentication and ConfigureAuthenticationV2 in startup.cs.
+        // This means that this code is resilient to whether or not the default authentication scheme is set in startup.
+        string scheme = EasyAuthAuthenticationDefaults.SWAAUTHSCHEME;
+        if (!_runtimeConfigProvider.IsLateConfigured)
+        {
+            AuthenticationOptions? dabAuthNOptions = _runtimeConfigProvider.GetConfig().Runtime?.Host?.Authentication;
+            scheme = ResolveConfiguredAuthNScheme(dabAuthNOptions?.Provider);
+        }
+
         // authNResult will be one of:
         // 1. Succeeded - Authenticated
         // 2. Failure - Token issue
         // 3. None - No token provided, no auth result.
-        AuthenticateResult authNResult = await httpContext.AuthenticateAsync();
+        AuthenticateResult authNResult = await httpContext.AuthenticateAsync(scheme);
 
         // Reject and terminate the request when an invalid token is provided
         // Write challenge response metadata (HTTP 401 Unauthorized response code
         // and www-authenticate headers) to the HTTP Context via JwtBearerHandler code
         // https://github.com/dotnet/aspnetcore/blob/3fe12b935c03138f76364dc877a7e069e254b5b2/src/Security/Authentication/JwtBearer/src/JwtBearerHandler.cs#L217
         if (authNResult.Failure is not null)
         {
-            await httpContext.ChallengeAsync();
+            await httpContext.ChallengeAsync(scheme);
             return;
         }
 
-        string clientDefinedRole = AuthorizationType.Anonymous.ToString();
+        // Manually set the httpContext.User to the Principal from the AuthenticateResult
+        // when we exclude setting a default authentication scheme in Startup.cs AddAuthentication().
+        // https://learn.microsoft.com/aspnet/core/security/authorization/limitingidentitybyscheme?view=aspnetcore-8.0
+        if (authNResult.Succeeded)
+        {
+            httpContext.User = authNResult.Principal;
```

Comment:
This setup of httpContext.User is the only required change for hot reload in this file,right? Is my understanding correct that the rest of the changes in this file are good to have but not really relevant to hot reload?

I dont mind the changes but trying to understand how the value of scheme affects hot reload..

Reply:
We need all the code in this file and on lines that precede this code. The following line is important because we must pass scheme to authenticateAsync to ensure the proper authentication handler is used. Scheme is determined using the runtime config. If the provider changes in runtime config, DAB must resolve that change and ensure the request is authenticated using the hotreloaded scheme.
```csharp
await httpContext.AuthenticateAsync(scheme);
```